account_asset_management/models/account_asset.py

```diff
@@ -0,0 +1,1092 @@
+
+    @api.model
+    def _default_company_id(self):
+        return self.env['res.company']._company_default_get('account.asset')
+
+    @api.multi
+    def _compute_move_line_check(self):
+        for asset in self:
+            for line in asset.depreciation_line_ids:
+                if line.move_id:
+                    asset.move_line_check = True
+                    continue
+
+    @api.depends('purchase_value', 'salvage_value', 'type', 'method')
+    @api.multi
+    def _compute_depreciation_base(self):
+        for asset in self:
+            if asset.type == 'view':
+                asset.depreciation_base = 0.0
+            elif asset.method in ['linear-limit', 'degr-limit']:
+                asset.depreciation_base = asset.purchase_value
+            else:
+                asset.depreciation_base = \
+                    asset.purchase_value - asset.salvage_value
+
+    @api.multi
+    @api.depends('type', 'depreciation_base',
+                 'depreciation_line_ids.type',
+                 'depreciation_line_ids.amount',
+                 'depreciation_line_ids.previous_id',
+                 'depreciation_line_ids.init_entry',
+                 'depreciation_line_ids.move_check',)
+    def _compute_depreciation(self):
+        for asset in self:
+            if asset.type == 'normal':
+                lines = asset.depreciation_line_ids.filtered(
+                    lambda l: l.type in ('depreciate', 'remove') and
+                    (l.init_entry or l.move_check))
+                value_depreciated = sum([l.amount for l in lines])
+                asset.value_residual = \
+                    asset.depreciation_base - value_depreciated
+                asset.value_depreciated = value_depreciated
+            else:
+                asset.value_residual = 0.0
+                asset.value_depreciated = 0.0
+
+    @api.multi
+    @api.constrains('parent_id')
+    def _check_recursion(self, parent=None):
+        res = super(AccountAsset, self)._check_recursion(parent=parent)
+        if not res:
+            raise UserError(
+                _("Error ! You can not create recursive assets."))
+        return res
+
+    @api.multi
+    @api.constrains('method', 'method_time')
+    def _check_method(self):
+        for asset in self:
+            if asset.method == 'degr-linear' and asset.method_time != 'year':
+                raise UserError(
+                    _("Degressive-Linear is only supported for Time Method = "
+                      "Year."))
+
+    @api.multi
+    @api.constrains('date_start', 'method_end', 'method_time')
+    def _check_dates(self):
+        for asset in self:
+            if asset.method_time == 'end':
+                if asset.method_end <= asset.date_start:
+                    raise UserError(
+                        _("The Start Date must precede the Ending Date."))
+
+    @api.onchange('purchase_value', 'salvage_value', 'date_start', 'method')
```

Comment:
lacks `depreciation_base`?

Reply:
Not needed, since `self.depreciation_base` is calculated inside the onchange and the triggering fields are the good ones. :wink: